ui/src/app/shared/components/resource-submit.tsx

```diff
@@ -7,6 +7,7 @@ interface ResourceSubmitProps<T> {
     defaultResource: T;
     resourceName: string;
     onSubmit: (value: T) => Promise<void>;
+    validate?: (values: T) => any;
```

Comment:
I'd recommend returning both a boolean and an optional string
```suggestion
    validate?: (values: T) => {valid: boolean, message?: string};
```